x/ecocredit/core/utils.go

```diff
@@ -269,3 +271,28 @@ func ExponentToPrefix(exponent uint32) (string, error) {
 	}
 	return e, nil
 }
+
+func CoinsToProtoCoins(coins sdk.Coins) []*basev1beta1.Coin {
+	result := make([]*basev1beta1.Coin, 0, coins.Len())
+	for _, coin := range coins {
+		result = append(result, &basev1beta1.Coin{
+			Denom:  coin.Denom,
+			Amount: coin.Amount.String(),
+		})
+	}
+
+	return result
+}
+
+func ProtoCoinsToCoins(coins []*basev1beta1.Coin) (sdk.Coins, bool) {
+	result := make([]sdk.Coin, 0, len(coins))
+	for _, coin := range coins {
+		amount, ok := sdk.NewIntFromString(coin.Amount)
+		if !ok {
+			return nil, ok
+		}
+		result = append(result, sdk.NewCoin(coin.Denom, amount))
+	}
+
+	return result, true
+}
```

Comment:
Ok to have these here but we may want to consider moving to the types package so that they can be reused outside the context of the ecocredit core submodule.

Reply:
+1 for moving to types. i think we should just do that now in this PR than later.